### ISSUE TITLE
Cache improvements

### ISSUE THREAD

**Issue by MolotovCherry** [nttld/setup-ndk#220]
Even right now although it adds it to the cache, it still has to download the whole NDK and can take about 2 minutes per run just downloading and extracting.

If the [official cache action](https://github.com/actions/cache) is supported, then we could use it before it before the setup-ndk action, and all this action would have to do is provide an input which indicates whether there was a cache hit or not, and add it to the PATH (rather than having to download and extract it all again). However, I believe some changes to where the NDK gets downloaded to might be required for the official cache action to cache it (not 100% sure about this) since the official cache action needs to know where to cache from, and this setup-ndk action currently extracts to a random identifier in the runner temp folder, so the official cache action doesn't really know *where* it is (also, I think the official cache action can only cache from the workspace directory, not completely sure though).

But other than that, supporting a run like this could cut times from about 2 minutes to 20 seconds.

An example API could be similar to this:
```yaml
  - uses: actions/cache@v2
    id: cache
    with:
      path: ndk
      key: ndk

  - uses: nttld/setup-ndk@v1
    id: setup-ndk
    with:
      ndk-version: r22b
      add-to-path: true
      directory: ndk #this is local to workspace root
      cache-hit: ${{ steps.cache.outputs.cache-hit }} # true or false
```
If `cache-hit` is true, then setup-ndk will only need to `add-to-path`, and can skip downloading, extracting, and add to tool cache steps.

**Comment by raftario** [comment]
I'll look into integrating with the cache action. At the moment the action is almost identical to the official `setup-xyz` actions and uses the SDK's tool cache, which in retrospect is not the best idea cause the chances of getting a runner that previously had NDK installed are really slim.
However last time I checked the SDK was about to get interfaces to the regular, non-tool cache, and if possible I'd rather use that as an alternative and keep the download locations and everything else standard.
How would you feel about that kind of solution ? To me it seems like it would be cleaner and simpler for end users but I might be missing something.

**Comment by cherryleafroad** [comment]
I decided to run some real world tests, [here#1](https://github.com/cherryleafroad/Testing/runs/3909467070?check_suite_focus=true) and [here#2](https://github.com/cherryleafroad/Testing/runs/3909506941?check_suite_focus=true). I found that indeed `actions/cache@v2` can cache the NDK path from `${{ runner.temp }}` and even extract back to it.

Saving the cache took a cool 3 minutes, but on cache hit next run it only took 43 seconds.

Compared to a run time of 1minute 30 seconds for the NDK tool to extract cache. However, it's noteworthy that NDK took finished in about 43 seconds and the rest of the time it was "adding to cache". So the "adding to cache" part seems to take the biggest hit from what I can tell.

Not sure whether this is a toss up or not. - May require further tests.

\* I heard there were some performance improvements to Linux and MacOS runners. It may be indeed that those are faster than the Windows runner was.
